agents/pfeiffer_tpg366/pfeiffer_tpg366_agent.py

```diff
@@ -0,0 +1,179 @@
+    pressure gauage controller
+    ip_address: ip address of the deivce
+    port: 8000 (fixed for the device)
+    
+    Attributes:
+       read_pressure reads the pressure from one channel (given as an argument)
+       read_pressure_all reads pressrue from the six channels
+       close closes the socket
+    
+    Todo:
+        -Generalize IP address inputs (input IP  via command line?)
+    """
+    def __init__(self, ip_address=IP_ADDRESS, port=8000, timeout=10, f_sample = 2.5):
+        self.ip_address = ip_address
+        self.port = port
+        self.comm = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.comm.connect((self.ip_address, self.port))
+        self.comm.settimeout(timeout)
+    
+    def read_pressure(self, ch_no):
+        """
+        Function to measure the pressure of one given channel
+        ch_no is the chanel to be measured (e.g. 1-6)
+        returns the measured pressure as a float
+        
+        Args:
+            ch_no: The channel to be measured (1-6)
+        Returns:
+            pressure as a float
+        """
+        msg = 'PR%d\r\n'%ch_no
+        self.comm.send(msg.encode())
+        status = self.comm.recv(BUFF_SIZE).decode()#Could probably use this to catch exemptions
+        self.comm.send(ENQ.encode())
+        read_str = self.comm.recv(BUFF_SIZE).decode()
+        pressure_str = read_str.split(',')[-1].split('\r')[0]
+        pressure = float(pressure_str)
+        #print(pressure)
+        return pressure        
+
+    def read_pressure_all(self):
+        """measure the pressure of all channel
+        Return an array of 6 pressure values as a float array
+
+        Args:
+            None
+        Returns:
+            6 element array corresponding to each channels pressure reading, as floats
+        """
+        msg = 'PRX\r\n'
+        self.comm.send(msg.encode())
+        status = self.comm.recv(BUFF_SIZE).decode()#Could probably use this to catch exemptions
+        self.comm.send(ENQ.encode())
+        read_str = self.comm.recv(BUFF_SIZE).decode()
+        pressure_str = read_str.split('\r')[0]
+        gauge_states = pressure_str.split(',')[::2]
+        gauge_states = np.array(gauge_states, dtype=int)#this is recorded just incase
+        pressures = pressure_str.split(',')[1::2]
+        pressures = [float(p) for p in pressures]
+        #print(pressures,type(pressures[0]),"This works")
+        return pressures 
+
+    def close(self):
+        """
+        Close the socket of the connection
+        """
+        self.comm.close()
+
+class pfeifferAgent:
+    def __init__(self, agent,ip_address = IP_ADDRESS, port=8000, f_sample=2.5):
+        self.active = True  
+        self.agent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+        self.f_sample = f_sample
+        self.take_data = False
+        self.module: Optional[Module] = None
+        self.gauge = pfeiffer()
+        self.port = port
```

Comment:
Similar to below, you should remove this line if you don't need to keep track of self.port. (This should probably also be passed to the pfeiffer object instantiation.

Reply:
You still don't need this class attribute, you can remove `self.port = port`.